### PR TITLE
Adding UP Inactviity Timer feature

### PR DESCRIPTION
**Pull request by vijay-hirani**
Added Branch to support User Plane Inactivity Timer Feature. Please review.

### CODE REVIEW

File: src/ggsn_gn.erl [hunk 1]
```diff
@@ -675,12 +682,19 @@ apply_context_change(NewContext0, OldContext, URRActions,
...
+
 %% 'Idle-Timeout' received from ergw_aaa Session takes precedence over configured one
-add_apn_timeout(Opts, Session, Context) ->
+add_apn_timeout(Opts, Session, Context, PCtx) ->
     SessionWithTimeout = maps:merge(maps:with(['Idle-Timeout'],Opts), Session),
     Timeout = maps:get('Idle-Timeout', SessionWithTimeout),
     ContextWithTimeout = Context#context{'Idle-Timeout' = Timeout},
-    {ContextWithTimeout, SessionWithTimeout}.
+    PCtxTimeout = map_to_up_inactivity_timer(Timeout),
+    PctxWithTimeout = PCtx#pfcp_ctx{up_inactivity_timer = PCtxTimeout},
```
Comment (RoadRunnr) [inline src/ggsn_gn.erl]: updating the pfcp_ctx should not be done in the protocol handlers. ergw_gsn_lib might be a better place to handle it.  The GTP context contains the timeout and is passed to gsn_lib, so all the values should be available.

Also beware that a idle timeout change might happen while the session running (NASREQ RAR can set it), so session modify needs to handle this as well.

File: src/ggsn_gn.erl [hunk 2]
```diff
@@ -675,12 +682,19 @@ apply_context_change(NewContext0, OldContext, URRActions,
     defer_usage_report(URRActions, UsageReport),
     Data#{context => NewContext, pfcp => PCtx}.
 
+map_to_up_inactivity_timer(SessionTimeout) when is_integer(SessionTimeout) ->
+    SessionTimeout div 1000; % UP timer measured in seconds
+map_to_up_inactivity_timer(_SessionTimeout) -> % CP Idle-Timeout infinty = 0 on UP
+    0. % Zero means no UP inactivity timer is activated
```
Comment (RoadRunnr) [inline src/ggsn_gn.erl]: undefined might be better

File: src/ggsn_gn.erl (hunk 1 above)
Comment (vijay-hirani) [reply src/ggsn_gn.erl]: Move the pfcp_ctx handling to ergw_gsn_lib.

File: src/ggsn_gn.erl (hunk 2 above)
Comment (vijay-hirani) [reply src/ggsn_gn.erl]: Changed handling of undefined scenario

**Comment by RoadRunnr** [comment]
superseded by #164

**Pull request closed without merging**